packages/ckeditor5-ui/src/dialog/dialogview.ts

```diff
@@ -0,0 +1,635 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/dialog/dialogview
+ */
+
+import {
+	type EventInfo,
+	type Locale,
+	type CollectionChangeEvent,
+	type DecoratedMethodEvent,
+	KeystrokeHandler,
+	FocusTracker,
+	Rect,
+	global,
+	toUnit
+} from '@ckeditor/ckeditor5-utils';
+import ViewCollection from '../viewcollection';
+import View from '../view';
+import FormHeaderView from '../formheader/formheaderview';
+import ButtonView from '../button/buttonview';
+import FocusCycler, { isViewWithFocusCycler, type FocusCyclerBackwardCycleEvent, type FocusCyclerForwardCycleEvent } from '../focuscycler';
+import DraggableViewMixin, { type DraggableView, type DraggableViewDragEvent } from '../bindings/draggableviewmixin';
+import DialogActionsView, { type DialogActionButtonDefinition } from './dialogactionsview';
+
+// @if CK_DEBUG_DIALOG // const RectDrawer = require( '@ckeditor/ckeditor5-utils/tests/_utils/rectdrawer' ).default;
+
+import cancelIcon from '@ckeditor/ckeditor5-core/theme/icons/cancel.svg';
+import type EditorUI from '../editorui/editorui';
+
+export enum DialogViewPosition {
+	SCREEN_CENTER = 'screen-center',
+	CURRENT_ROOT_CENTER = 'current-root-center',
+	CURRENT_ROOT_NE = 'current-root-ne',
+	CURRENT_ROOT_NW = 'current-root-nw'
+}
+
+const toPx = toUnit( 'px' );
+
+/**
+ * TODO
+ */
+export default class DialogView extends DraggableViewMixin( View ) implements DraggableView {
```

Comment:
Same as with `DialogsActionsView`, please check which properties must be public.

Reply:
Regarding the first comment - I think it was made on purpose. `extends DraggableViewMixin( View )` part provides the dragging mechanisms (and the whole `View` core class) to the class and doesn't expect anything in return. OTOH the `implements DraggableView` indicates that there are some additional constraints (namely, some props/methods that need to be defined) in the `DialogView` class. And I think @oleq wanted to emphasise this.

I think it looks a bit strange right now as this interface is rather small:

[https://github.com/ckeditor/ckeditor5/pull/15285/files#diff-2434be37d8a50dd90e47ad6ed9e54254b4745e7e951b9968b03cb4c83e1ff093R160-R164](https://github.com/ckeditor/ckeditor5/pull/15285/files#diff-2434be37d8a50dd90e47ad6ed9e54254b4745e7e951b9968b03cb4c83e1ff093R160-R164)

with `isDraggable` to be gone, and `resetDrag()` not even implemented (btw I wonder why it works :thinking: ). But I think it may still make sense to keep these two constructs separate.

Reply:
As for the properties accessibility, I'm rather cautious about it due to [https://github.com/ckeditor/ckeditor5/pull/15285#discussion_r1400475486](https://github.com/ckeditor/ckeditor5/pull/15285#discussion_r1400475486) - access to stuff like `keystrokes` property may be expected for the integrator creating their own dialogs. And some properties that could be private are observables (e.g. `DialogView#_top`) so they have to stay public. So I did some changes but only those that seemed safe (both in `DialogView` and `DialogActionsView`).